x-pack/plugins/observability_solution/infra/public/pages/metrics/hosts/components/kpis/kpi.tsx

```diff
@@ -4,22 +4,21 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import React, { useMemo } from 'react';
-import type { LensConfig, LensDataviewDataset } from '@kbn/lens-embeddable-utils/config_builder';
-import { useDataView } from '../../../../../hooks/use_data_view';
-import { METRICS_TOOLTIP } from '../../../../../common/visualizations';
-import { LensChart, TooltipContent } from '../../../../../components/lens';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { HostKpiCharts } from '../../../../../components/asset_details';
 import { buildCombinedHostsFilter } from '../../../../../utils/filters/build';
 import { useUnifiedSearchContext } from '../../hooks/use_unified_search';
 import { useHostsViewContext } from '../../hooks/use_hosts_view';
 import { useHostCountContext } from '../../hooks/use_host_count';
 import { useAfterLoadedState } from '../../hooks/use_after_loaded_state';
+import { useMetricsDataViewContext } from '../../hooks/use_metrics_data_view';
 
-export const Kpi = ({ id, height, ...chartProps }: LensConfig & { height: number; id: string }) => {
+export const KpiCharts = () => {
   const { searchCriteria } = useUnifiedSearchContext();
   const { hostNodes, loading: hostsLoading, searchSessionId } = useHostsViewContext();
-  const { isRequestRunning: hostCountLoading } = useHostCountContext();
-  const { dataView } = useDataView({ index: (chartProps.dataset as LensDataviewDataset)?.index });
+  const { isRequestRunning: hostCountLoading, data: hostCountData } = useHostCountContext();
+  const { dataView } = useMetricsDataViewContext();
```

Comment:
I tried to test locally with oblt cluster data and I see infinite loading for the Logs chart:

https://github.com/elastic/kibana/assets/14139027/e532e7a0-f844-4c4c-888c-099f00eb3558


I checked the changes in this PR several times I am not sure if the issue comes from it but I think maybe because the logs chart is using the logs dataview it could be around that change. In general, it should be an error in case my env is misconfigured or the logs are not available there (for example) - so it could be also something around the error handling. I checked on oblt cluster and the same host looks fine there:
<img width="1920" alt="image" src="https://github.com/elastic/kibana/assets/14139027/5c09dbd2-d256-4939-a163-341db186e143">

Reply:
I couldn't reproduce the problem. 

<img width="805" alt="image" src="https://github.com/elastic/kibana/assets/2767137/f05e9290-0cb6-4dc2-9897-c97cae12d544">

When there is an error in the logs indices config, this is usually what happens:

<img width="874" alt="image" src="https://github.com/elastic/kibana/assets/2767137/7b9b9b1c-0e96-45ff-9330-46f416d2c0d9">

I'm confused whether you think the endless loading problem is caused by the `kpi` component